compiler/src/dotty/tools/dotc/core/Uniques.scala

```diff
@@ -10,7 +10,9 @@ import scala.annotation.tailrec
 
 class Uniques extends WeakHashSet[Type](Config.initialUniquesCapacity):
   override def hash(x: Type): Int = x.hash
-  override def isEqual(x: Type, y: Type) = x.eql(y)
+  override def isEqual(x: Type | Null, y: Type | Null) =
```

Comment:
Can we make isEqual only take non-null values? (both here and in WeakHashSet)? I'd prefer that.

Reply:
Done